.github/workflows/puppetize.yml

```diff
@@ -36,10 +38,23 @@ jobs:
         run: |
           Import-Module -Name PSDesiredStateConfiguration -Force
           Import-Module -Name ./src/BuildMatrix/BuildMatrix.psd1 -Force
+          # Get the top 50 downloaded DSC modules from the forge
+          $response = Invoke-WebRequest -URI https://forgeapi.puppet.com/v3/modules?owner=dsc`&limit=50`&endorsements=approved`&sort_by=downloads -Method Get -UseBasicParsing
+          $parsed = $response.Content | ConvertFrom-Json
+          $fileName = "./dsc_resources.yml"
+
+          # Add the module to the dsc_resource.yml if it doesn't exist
+          ForEach ($Module in $parsed.results.name) {
+            $module_name_exists =  Select-String -Quiet -Pattern $Module -Path $fileName
```

Comment:
@jordanbreen28: Does it matter if Microsoft uplifts the dsc resource version, e.g., adding or removing a parameter?  Would this potentially need to be regenerated?  If we're only checking for the absence of the resource in dsc_resources.yml.

Reply:
hey @gavindidrichsen apologies not sure I follow..
So we use the `dsc_resources.yml` file [here](https://github.com/puppetlabs/Puppet.Dsc/blob/main/dsc_resources.yml) to store DSC modules which regularly need to be puppetized. So essentially what this step does is:

1. Polls the Forge API for the top 50 downloaded Puppet DSC Modules from forge.puppet.com
2. Adds each module to `dsc_resources.yml` if the module _is not_ already present in the file
3. This file is then passed to `puppetize.yml`, which will puppetize (i.e. create the puppet module) for each of those DSC modules which has had a new version released to the powershell gallery since the last time it was puppetized.

Hope that answers your question - or give me a ping and can explain it some more if i missed your point completely!

Reply:
Thanks for the clarification @jordanbreen28!